### PR TITLE
Update Studio API docs for JWT

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/4429

### CODE REVIEW

File: source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst [hunk 1]
```diff
@@ -10,71 +10,113 @@ Working with Crafter Studio's APIs
...
-As we'll see below, the Login API sends a header and a cookie CSRF/XSRF token value which must also be sent in subsequent calls. The value is arbitrary and is dictated by the caller. Strong values are more secure than predictable weak ones.
+#. **Authentication**
 
-Let's begin:
+   There are two authentication methods that can be used: JWT and HTTP Basic
 
-#. **Authenticate**
+   JWT is standard widely adopted as a secure authentication method, this is the prefered way. HTTP Basic is
+   simple but not as secure as JWT and for that reason it is disabled by default, however, it can be very useful for
+   local development environments.
```
Comment (alhambrav) [inline source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst:29]: For the above paragraph, a typo and maybe reword to something like this:

```
JWT is a widely adopted standard for secure authentication and is the preferred way.  

HTTP Basic is simple but not as secure as JWT and for that reason it is disabled by default.  However, it can be very useful for local development environments
```

File: source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst [hunk 2]
```diff
@@ -10,71 +10,113 @@ Working with Crafter Studio's APIs
...
+   the operations that you need to execute. For this example we will generate a token for the ``admin`` user but in
+   a real scenario you should create a user with the least amount of permissions possible.
 
-   .. code-block:: bash
+   .. note::
 
-       curl -d '{"username":"admin","password":"admin"}' --cookie "XSRF-TOKEN=A_VALUE" --header "X-XSRF-TOKEN:A_VALUE" --header "Content-Type: application/json" -v -X POST http://localhost:8080/studio/api/1/services/api/1/security/login.json
+     Not all users have the required permission ``manage_access_token`` to create access tokens, if the
+     ``Token Management`` tool is not shown in the global menu you will need to request the permission or an
+     access token to a system administrator.
```
Comment (alhambrav) [inline source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst:39]: A typo and maybe break it up a little?

```
Not all users have the required permission ``manage_access_token`` to create access tokens.   
If the ``Token Management`` tool is not shown in the global menu you will need to request the permission for an
     access token from a system administrator.
```

File: source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst [hunk 3]
```diff
@@ -10,71 +10,113 @@ Working with Crafter Studio's APIs
...
-      < Transfer-Encoding: chunked
-      < Date: Mon, 22 Jan 2018 21:32:48 GMT
-      <
-      * Connection #0 to host localhost left intact
-      {"username":"admin","first_name":"admin","last_name":"admin","email":"evaladmin@example.com"}
-
-   Note the response returned is a successful 200 status code and the response contains JSON with details for the authenticated user.
-
-   Also found as part of the request is the ``JSESSIONID`` cookie.  You will need this value for all future requests.
+   To use JWT you need to obtain a an access token following these steps:
```
Comment (alhambrav) [inline source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst:43]: Typo? Remove the `a` before `an`

File: source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst [hunk 4]
```diff
@@ -10,71 +10,113 @@ Working with Crafter Studio's APIs
...
+      .. figure:: /_static/images/jwt/token-management.png
+        :width: 70%
+        :alt: Crafter Studio - Token Management
+        :align: center
+      |
+
+   3. Create a new access token
+
+      The only required field for the access token is the label to identify it, however, it is recommended to also
+      set an expiration date to minimize the risk of lost or stolen tokens being used without noticing.
```
Comment (alhambrav) [inline source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst:64]: Maybe say, 
```
it is also recommended to set an expiration date to minimize the risk of lost or stolen tokens being used without being noticed.
```

File: source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst [hunk 5]
```diff
@@ -10,71 +10,113 @@ Working with Crafter Studio's APIs
...
+      Most HTTP clients provide built-in support for access token athentication, in the case of curl you will need
+      to use the header option ``--header`` or ``-H``
+
+      ``curl -H 'Authentication: Bearer <access token>' ...``
+
+   To use HTTP Basic you need to follow these steps:
+
+   1. Enable HTTP Basic auth and restart Crafter Studio
+
+      Update the Crafter Studio configuration override (either from the disk or from the UI) to add the following:
```
Comment (alhambrav) [inline source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst:105]: Maybe say the location of the override file on disk (e.g. `CRAFTER_HOME/data/repos/global/configuration/studio-config-override.yaml`) and also how to access from the UI (e.g. From the global menu -> `Global Config`)

File: source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst [hunk 6]
```diff
@@ -10,71 +10,113 @@ Working with Crafter Studio's APIs
...
+      .. code-block:: yaml
+
+        # Indicates if HTTP Basic auth should be enabled for the API
+        studio.security.http.basic.enabled: true
+      |
+
+
+   2. Include the credentials with all requests
+
+      Most HTTP clients provide an easy way to use HTTP Basic, in the case of curl you can simple use the user options
```
Comment (alhambrav) [inline source/developers/cook-books/how-tos/working-with-crafter-studios-api.rst:116]: Typo? `simply` instead of `simple`

**Review by alhambrav — CHANGES_REQUESTED** [review]

**Review by alhambrav — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]
